packages/vite/src/node/plugins/css.ts

```diff
@@ -116,13 +140,44 @@ export interface CSSModulesOptions {
       ) => string)
 }
 
+export type ResolvedCSSOptions = Omit<CSSOptions, 'lightningcss'> & {
+  lightningcss?: {
+    targets: LightningCSS['Targets']
+    modules: LightningCSS['CSSModulesConfig'] | undefined
+    drafts: LightningCSS['Drafts']
+  }
+}
+
+export function resolveCSSOptions(
+  options: CSSOptions | undefined,
+  resolvedBuildOptions: ResolvedBuildOptions,
+): ResolvedCSSOptions | undefined {
+  if (options?.lightningcss) {
+    return {
+      ...options,
+      lightningcss: {
+        targets:
+          resolvedBuildOptions.cssMinifier === 'lightningcss' &&
+          resolvedBuildOptions.lightningcss?.targets
+            ? resolvedBuildOptions.lightningcss.targets
+            : convertTargets(resolvedBuildOptions.cssTarget),
```

Comment:
We need to provide a way to downgrade a bit in dev because otherwise if you use CSS nesting, you will not be able to test your your in Firefox in dev. 

And for TS we do force a target level of ~ES2020 in dev IIRC.

We could do something similar where we only slightly downgrade, but I think this is more stable to run the full downgrade when you start applying one.

Reply:
Hmm yeah I think you're right about the TS part, since I'm thinking of this code where it sets `esnext` as fallback: https://github.com/vitejs/vite/blob/ff3ce312a572ec126808afb97b5d3f0a6f9adcb1/packages/vite/src/node/plugins/esbuild.ts#L219

But in general, Vite would only use the `build` options for builds (AFAIK). For dev, the user need to use other (dev/build agnostic) options like `esbuild.target` and `tsconfig.json#target`. For CSS, I don't think we have meddled with up/down-levelling before.

Maybe we can move the configuration to `css.lightningcss.target` instead? (matching `esbuild.target`) 🤔

Reply:
FWI: We decided to keep css.lightningcss.target only for dev with a default to ~es2020 and for build we don't expose the possibility of setting lightningcss targets and always convert from the 'esbuild' one.